Project/DAQ_System/Program/Src/app.cpp

```diff
@@ -93,6 +104,9 @@ void cppMain() {
 	std::unique_ptr<sensor::ILinearPotentiometer> lin_pot(nullptr);
 	lin_pot = std::make_unique<sensor::SLS1322>(hadc1);
 
+	IGyroscope *l3gd20h = new Gyroscope_L3GD20H();
+
```

Comment:
Great! This is exactly what we need. However, raw pointers require manual deletion. Also, for our use case, we do not want multiple pointers (aka multiple, or shared references) to the object being pointed to.

Smart Pointers are a thin "wrapper" class. In other words, smart pointers are C++ objects that have a **raw** pointer inside it — so they are a little bit nicer to work with and do not require manual deletion of the object.

`unique_ptr` is a smart pointer that allows only one pointer to the object.

To use it, try the following:
```c++
std::unique_ptr<sensor::IGyroscope> gyroscope(nullptr);      // Declare an abstract pointer
gyroscope = std::make_unique<sensor::L3GD20H>(hi2c1);   // Create an object and assign it to the pointer
```

Notice that I named the pointer `gyroscope` instead of `l3gd20h`. This is because the abstract interface pointer should not care about which gyroscope is being used, as long as it satisfies the rules of the `IGyroscope` interface.